contracts/misc/NFTFloorOracle.sol

```diff
@@ -305,37 +293,27 @@ contract NFTFloorOracle is Initializable, AccessControl, INFTFloorOracle {
             return (true, twap);
         }
         //use memory here so allocate with maximum length
-        uint256[] memory validPriceList = new uint256[](
-            feeders.length * MAX_SUBMISSION
-        );
+        uint256[] memory validPriceList = new uint256[](feeders.length);
         uint256 validNum = 0;
         //aggeregate with price in each ring position of all feeders
-        for (uint256 i = 0; i < MAX_SUBMISSION; i++) {
-            for (uint256 j = 0; j < feeders.length; j++) {
-                PriceInformationBuffer memory feederBuffer = feederRegistrar
-                    .feederBuffer[feeders[j]];
-                PriceInformation memory priceInfo = feederBuffer.ring[i];
-                if (priceInfo.lastUpdateTime > 0) {
-                    uint256 diffTime = currentTime - priceInfo.lastUpdateTime;
-                    if (diffTime < config.expirationPeriod) {
-                        validPriceList[validNum] = priceInfo.twap;
-                        validNum++;
-                    }
+        for (uint256 j = 0; j < feeders.length; j++) {
+            PriceInformation memory priceInfo = feederRegistrar.feederPrice[
+                feeders[j]
+            ];
+            if (priceInfo.lastUpdateTime > 0) {
+                uint256 diffTime = currentTime - priceInfo.lastUpdateTime;
+                if (diffTime < config.expirationPeriod) {
+                    validPriceList[validNum] = priceInfo.twap;
+                    validNum++;
                 }
             }
-            //break earlier if we have enough valid prices
-            if (validNum >= feeders.length) {
-                break;
-            }
         }
-
-        // only calculate median value when number of valid data greater than minCountToAggregate
-        if (validNum >= feeders.length) {
-            // ignore sort for saving gas, we just pick fixed element from validPriceList
-            return (true, validPriceList[validNum / 2]);
+        //we allow at most 1 feeder down
+        if (validNum < (feeders.length - 1)) {
```

Comment:
use validPriceList.length

Reply:
`validPriceList.length==feeders.length` we leave room for 1 oracle down here

Reply:
change to 2/3 as threshold,  need to confirm with team @yz89

Reply:
use fixed value as threshold instead